tools/admin_ui_selenium_test.py

```
@@ -0,0 +1,393 @@
+        if len(buttons) < 2:
+            return None
+        return buttons[-1] if next_page else buttons[0]
+
+    def move_grid_to_first_page(grid_id: str) -> None:
+        while True:
+            prev_button = grid_edge_button(driver, grid_id, next_page=False)
+            if prev_button is None or not prev_button.is_enabled():
+                return
+            before = current_grid_page_marker(driver, grid_id)
+            prev_button.click()
+            wait.until(lambda d: current_grid_page_marker(d, grid_id) != before)
```

Ruff (0.15.6)

[warning] Function definition does not bind loop variable `before`

(B023)